### ISSUE TITLE
id-match should ignore named imports

### ISSUE THREAD

**Issue by gajus** [eslint/eslint#14005]
**The version of ESLint you are using.**

v7.17.0

**The problem you want to solve.**

Import named exports without needing to use eslint-disable

**Your take on the correct solution to problem.**

https://eslint.org/docs/rules/id-match should have an option `ignoreNamedImport` that would suppress id-match errors when import variables.

e.g.

```js
import {
  stubIsValidRecaptchaV2Response,
} from '../stubs';
```

Here I am getting id-match error, but I've already suppressed this error in the file where the variable is declared.

**Are you willing to submit a pull request to implement this change?**

No

**Comment by Yash-Singh1** [comment]
The `camelcase` rule already implements, this with a `ignoreImports` option. For consistency, the option name should also be `ignoreImports`.

**Comment by anikethsaha** [comment]
Thanks for the issue

I think this makes sense to have an option. +1 for this. 
If we are naming it as `ignoreImports` then I guess it would check for default imports too (as per the name).

IMO, I would have this option just for named imports alone as in default we do have the control over the identifier name (we can use local id name for named imports too though )

Lets see if others have better thoughts in this

**Comment by nzakas** [comment]
Oops, looked like we lost track of this. What does everyone think? @eslint/eslint-tsc

**Comment by mdjermanovic** [comment]
I think this falls under [our policies](https://github.com/eslint/eslint#stylistic-rule-updates) of not adding new options to stylistic rules, so in my opinion the best course of action would be to make a custom rule/plugin.

**Comment by nzakas** [comment]
@mdjermanovic I agree.

closing per previous comment.
